.github/workflows/shared-wildfly-build-and-test.yml

```diff
@@ -0,0 +1,98 @@
+name: Build and Test WildFly
+
+on:
+  workflow_call:
+    inputs:
+      test-arguments:
+        description: "Test arguments"
+        required: false
+        default: ""
+        type: string
+      build-arguments:
+        description: "Build arguments"
+        required: true
+        default: ""
+        type: string
+      maven-repo-path:
+        description: "Maven repo path to extract to local cache"
+        required: true
+        default: ""
+        type: string
+      maven-repo-name:
+        description: "Maven repo name to extract to local cache"
+        required: true
+        default: ""
+        type: string
+      java-versions:
+        description: "JDK versions array"
+        required: true
+        default: ""
+        type: string
+      os:
+        description: "OS array"
+        required: true
+        default: ""
+        type: string
+
+jobs:
+  wildfly-build-and-test:
+    name: ${{ matrix.os }}-jdk${{ matrix.java }} ${{ inputs.test-arguments }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ${{ fromJson(inputs.os) }}
+        java: ${{ fromJson(inputs.java-versions) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: wildfly/wildfly
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'temurin'
+          cache: 'maven'
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.maven-repo-name }}
+          path: .
+      - name: Extract Maven Repo
+        shell: bash
+        run: |
+         tar -xzf ${{ inputs.maven-repo-path }} -C ~
+      - name: Build and Test WildFly SNAPSHOT ${{ inputs.build-arguments }}  ${{ inputs.test-arguments }}
+        run: mvn -U -B -ntp clean install ${{ inputs.build-arguments }} ${{ inputs.test-arguments }}
+        shell: bash
+        # We need this due to long file names on Windows: https://github.com/actions/upload-artifact/issues/240
+      - name: Find reports
+        if: failure()
+        run: |
+          {
+            echo 'SUREFIRE_REPORTS<<EOF'
+            find . -path '**/surefire-reports/*.txt'
```

Comment:
Minor, but we should leave the *.txt off IMO. I've been capturing the `.xml` files too as they can be helpful. I've written a tool that parses them and gives information to easily get data from the failed tests.

Reply:
Done.